src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -162,6 +163,69 @@ def create_random_clustered_circles_matrix(self, con_type: ConnectionTypes, circ
                                           p=[daily_connections_float / total_connections_float, weekly_connections_float / total_connections_float])
             v = np.full_like(conns, strengthes, dtype=np.float32)
             self.matrix[depth, agent.index, conns] = v
+            
+    def _create_geographically_clustered_circles_matrix(self, con_type: ConnectionTypes, circles: List[SocialCircle], depth):
```

Comment:
there was no reference to the geographic circles.
for each node, there should be a higher chance (given as an input) to select a node from your own geographic circle.
this way, you would get a connections graph clustered by geographic circles.

Reply:
renamed